buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1139,7 +1140,23 @@ artifactId=${project.name}
         outputs.upToDateWhen { false }
 
         include "**/*IT.class"
-        systemProperties.beamTestPipelineOptions = configuration.integrationTestPipelineOptions
+
+        def pipelineOptionsString = configuration.integrationTestPipelineOptions
+        if(pipelineOptionsString && configuration.runner?.equalsIgnoreCase('dataflow')) {
+          project.evaluationDependsOn(":beam-runners-google-cloud-dataflow-java-legacy-worker")
+          def allOptionsList = (new JsonSlurper()).parseText(pipelineOptionsString)
+          def dataflowWorkerJar = project.findProperty('dataflowWorkerJar') ?:
+                  project.project(":beam-runners-google-cloud-dataflow-java-legacy-worker").shadowJar.archivePath
+
+          allOptionsList.addAll([
```

Comment:
What will happen when we already passed such pipeline options before (eg. --dataflowWorkerJar pointing to some other location)? Will the option be overwritten/duplicated?

Reply:
I assume if there are duplicated args/options, the place inside pipeline that processes all the PipelineOptions will take care of the duplication. It would be too much redundancy if we need to do that in every gradle task. Correct me if I am wrong.

Reply:
Duplicate args will result in an error when being parsed later. In this case we will expect that the list will need deduplication/special handling built into either the Jenkins groovy job definition or within the Gradle code here to not add args that are effectively duplicated.

I think we can handle this case if it ever becomes an issue at a later point in time.

Reply:
No, duplicate options will be an error during parsing.

I don't think its worthwhile investing the effort to deduplicate here though. We can add/update the logic if it is ever needed.

Reply:
I agree with Luke. It's easy to fix in another contribution when needed.